version: '3.3'
services:
  postgres:
    image: postgres:alpine
    container_name: postgres
    ports:
      - 5432:5432
    volumes:
      - postgres:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=postgres
      - PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
      - LANG=en_US.utf8
      - PG_MAJOR=15
      - PG_VERSION=15.4
      - PG_SHA256=baec5a4bdc4437336653b6cb5d9ed89be5bd5c0c58b94e0becee0a999e63c8f9
      - DOCKER_PG_LLVM_DEPS=llvm15-dev \t\tclang15
      - PGDATA=/var/lib/postgresql/data

  redis:
    image: redis/redis-stack-server:latest
    container_name: redis
    restart: unless-stopped
    ports:
      - "6379:6379"
    volumes:
      - redis:/data
    healthcheck:
      test:
        [
          "CMD",
          "redis-cli",
          "-h",
          "localhost",
          "-p",
          "6379",
          "ping"
        ]
      interval: 2s
      timeout: 1m30s
      retries: 5
      start_period: 5s
    
volumes:
  postgres:
    external: true
    name: 93b53676129869b8e9725bf25a216387dd6d594f1141dd797fe8f41e0f499f37

  redis:
    driver: local